### ISSUE TITLE
Unable to use Ref when _id type is not ObjectID

### ISSUE THREAD

**Issue by shlomokraus** [szokodiakos/typegoose#181]
For my schema, I am using shortId to generate an id for user. 

```js
class User extends Typegoose {
   @prop({default: shortid.generate})
   _id!: string;
}

class Car extends Typegoose {
  @prop({ref: User})
  owner: Ref<User>
}

//  given we have a user with this id
const car = new CarModel({owner: 'abcdef'})
```

This worked well with mongoose, but when I try to Ref a prop to User I get validation error: 
```
    ValidationError: owner: Cast to ObjectID failed for value "rJXP68_UX" at path "owner"
```

Is it possible to use refs with string types?

**Comment by nihiluis** [comment]
I'd like to know/suggest this too.

https://github.com/szokodiakos/typegoose/pull/149

**Comment by hasezoey** [comment]
>     ValidationError: owner: Cast to ObjectID failed for value "rJXP68_UX" at path "owner"

not an error from typegoose

> Is it possible to use refs with string types?

should be possible

**Comment by hasezoey** [comment]
sorry, my previous comment was wrong, thanks to @mfulop you can now use types that are *not* ObjectID

@Ben305 this can be closed as "solved" - thanks to #369
